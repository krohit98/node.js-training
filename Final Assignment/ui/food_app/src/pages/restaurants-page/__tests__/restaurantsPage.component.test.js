import { render, screen } from '@testing-library/react';
import ShallowRenderer from 'react-test-renderer/shallow';
import RestaurantsPage from '../restaurantsPage.component';
import RestaurantTable from '../../../components/restaurant-table/restaurantTable.component';

jest.mock('../../../components/restaurant-table/restaurantTable.component');

describe("Unit testing RestaurantsPage component",()=>{

    test("Snapshot testing",()=>{
        const renderer = new ShallowRenderer();
        renderer.render(<RestaurantsPage />);
        const result = renderer.getRenderOutput();

        expect(result).toMatchSnapshot();
    })

    test("Should render RestaurantsPage correctly",()=>{
        RestaurantTable.mockImplementation(()=><div>RestaurantTableComponent</div>);
        render(<RestaurantsPage />);
        const heading = screen.getByText("List of Places");
        const restaurantTable = screen.getByText("RestaurantTableComponent");
        expect(heading).toBeInTheDocument();
        expect(restaurantTable).toBeInTheDocument();
    })
})