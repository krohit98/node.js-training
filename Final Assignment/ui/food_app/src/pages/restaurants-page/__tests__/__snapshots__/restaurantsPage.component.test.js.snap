// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit testing RestaurantsPage component Snapshot testing 1`] = `
<div
  className="container pageDiv"
>
  <h1
    className="display-2 pageHeading"
  >
    List of Places
  </h1>
  <RestaurantTable />
</div>
`;
